src/thriveopedia/Thriveopedia.cs

```diff
@@ -411,6 +452,28 @@ private void OnSearchUpdated(string newText)
         }
     }
 
+    /// <summary>
+    ///   Recursively gets all descendents of this page in the page tree.
+    /// </summary>
+    private List<ThriveopediaPage> GetAllChildren(ThriveopediaPage page)
```

Comment:
This creates a bunch of temporary lists. I think a better solution would be to return an IEnumerable and enumerate the child pages like:

```
foreach(var directChild in ...)
{
  foreach (var descendant in GetAllChildren(directChild)
  {
      yield return descendant;
  }
}
```

Reply:
Changed to this approach which (after adding `yield return directChild` as well) seems to behave the same.